### PR TITLE
POL-242 New Resource Table 4

### PR DESCRIPTION
**Pull request by SheikSena**
### Description

New Resource Table Changes

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Comment by SheikSena** [comment]
@cdwilhelm I'll fix those 2 issues .

File: cost/aws/rds_instance_cloudwatch_utilization/rds_instance_cloudwatch_utilization.pt [hunk 1]
```diff
@@ -398,19 +413,16 @@ end
...
-    label "Escalation approval"
-    description "Approve RDS RightSizing action"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for approval"
-      description "Explain why you are approving the action"
-    end
-  end
+  automatic false
+  label "Approve RDS RightSizing"
```
Comment (cdwilhelm) [inline cost/aws/rds_instance_cloudwatch_utilization/rds_instance_cloudwatch_utilization.pt:424]: Change to Update Instance

File: security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt [hunk 2]
```diff
@@ -262,19 +272,16 @@ end
...
-    label "Approve Resource Deletion"
-    description "Approve escalation to run RightScale Cloud Workflow to delete Internet-facing ELB's & ALB's"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for Approval"
-      description "Explain why you are approving the action"
-    end
-  end
+  automatic false
+  label "Delete Internet-facing ELB's & ALB's"
```
Comment (cdwilhelm) [inline security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt:283]: Change to Delete Resource, or Delete ELB

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt [hunk 3]
```diff
@@ -245,59 +245,66 @@ end
...
-    label "Approve Resource Deletion"
-    description "Approve escalation to run RightScale Cloud Workflow to delete publicly accessible RDS-instances"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for Approval"
-      description "Note: RDS Instances with 'DB Instance Status' other than 'Available' state and instances with delete Protection enabled cannot be deleted"
-    end
-  end
+  automatic false
+  label "Delete Publicly Accessible RDS-Instances"
```
Comment (cdwilhelm) [inline security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt:306]: Change to Delete Instance

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt [hunk 4]
```diff
@@ -245,59 +245,66 @@ end
...
-    label "Approve RDS Public Access Rule Config Change"
-    description "Approve escalation to run RightScale Cloud Workflow to public access rule config change for RDS instances"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for Approval"
-      description "Note: 'Public Access Config' will be modified for the RDS instances that has 'Available' state and can not be modified 'public access config' for other states like stopped, stopping & modifying etc.."
-    end
-  end
+  automatic false
+  label "Approve RDS Public Access Rule Configuration Change"
```
Comment (cdwilhelm) [inline security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt:299]: Change to Update Instance

File: security/storage/aws/s3_buckets_without_server_access_logging/aws_s3_buckets_without_server_access_logging.pt [hunk 5]
```diff
@@ -182,37 +182,39 @@ end
...
-    label "Approve Enable Bucket Logging"
-    description "Approve escalation to run RightScale Cloud Workflow to enable bucket logging"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for Approval"
-      description "Explain why you are approving the action"
-    end
-  end
+  automatic false
+  label "Approve To Enable Bucket Logging"
```
Comment (cdwilhelm) [inline security/storage/aws/s3_buckets_without_server_access_logging/aws_s3_buckets_without_server_access_logging.pt:216]: Change to Update Bucket.

Comment (SheikSena) [reply security/storage/aws/s3_buckets_without_server_access_logging/aws_s3_buckets_without_server_access_logging.pt]: Done !

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt (hunk 4 above)
Comment (SheikSena) [reply security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt]: Done !

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt (hunk 3 above)
Comment (SheikSena) [reply security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt]: Done !

File: security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt (hunk 2 above)
Comment (SheikSena) [reply security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt]: Done !

File: cost/aws/rds_instance_cloudwatch_utilization/rds_instance_cloudwatch_utilization.pt (hunk 1 above)
Comment (SheikSena) [reply cost/aws/rds_instance_cloudwatch_utilization/rds_instance_cloudwatch_utilization.pt]: Done !

**Comment by cdwilhelm** [comment]
@SheikSena  can you please fix the conflict in this PR

File: operational/aws/long_running_instances/long_running_instances.pt [hunk 6]
```diff
@@ -246,9 +253,9 @@ escalation "email" do
 end
 
 escalation "esc_stop_or_terminate" do
-  automatic false
+  automatic contains($param_automatic_action, "Stop/Terminate Instances")
```
Comment (cdwilhelm) [inline operational/aws/long_running_instances/long_running_instances.pt:256]: It looks like this policy as two actions, stop or terminate.  But the param_automatic_action only has one value.  There is also another `action` parameter that should be removed.

File: operational/aws/long_running_instances/README.md [hunk 7]
```diff
@@ -17,6 +17,10 @@ the policy can be applied.
 - *Days Old* - Number of days to be running before included in list.
 - *Action to Take* - Either Stop or Terminate the instance
 - *Exclusion Tag Key:Value* - Cloud native tag key to ignore instances. Format: Key:Value
+- *Automatic Actions* - When this value is set, this policy will automatically take the selected action(s).
```
Comment (cdwilhelm) [inline operational/aws/long_running_instances/README.md:20]: Remove the Action to take parameter

File: operational/aws/long_running_instances/long_running_instances.pt [hunk 8]
```diff
@@ -44,6 +44,13 @@ parameter "param_exclude_tags" do
   allowed_pattern /(^$)|([\w]?)+\:([\w]?)+/
 end
 
+parameter "param_automatic_action" do
+  type "list"
+  label "Automatic Actions"
+  description "When this value is set, this policy will automatically take the selected action(s)"
+  allowed_values ["Stop/Terminate Instances"]
```
Comment (cdwilhelm) [inline operational/aws/long_running_instances/long_running_instances.pt:51]: there should be two values, stop and terminate.

File: operational/aws/long_running_instances/long_running_instances.pt (hunk 6 above)
Comment (cdwilhelm) [inline operational/aws/long_running_instances/long_running_instances.pt:256]: there should be two actions, stop or terminate

**Comment by SheikSena** [comment]
@cdwilhelm, The action parameter (Stop, Terminate Instance) we need I guess. If we remove that parameter and add those values in automatic parameter (STOP, TERMINATE) since it is multi-select there is a chance that the user might select both the options (STOP & TERMINATE Instance) then we can't decide which one to execute. let me know your thoughts on this. Thanks

**Comment by cdwilhelm** [comment]
> @cdwilhelm, The action parameter (Stop, Terminate Instance) we need I guess. If we remove that parameter and add those values in automatic parameter (STOP, TERMINATE) since it is multi-select there is a chance that the user might select both the options (STOP & TERMINATE Instance) then we can't decide which one to execute. let me know your thoughts on this. Thanks

@SheikSena   can we make it a string instead of list and let the user only select one of the options?

**Comment by SheikSena** [comment]
> > @cdwilhelm, The action parameter (Stop, Terminate Instance) we need I guess. If we remove that parameter and add those values in automatic parameter (STOP, TERMINATE) since it is multi-select there is a chance that the user might select both the options (STOP & TERMINATE Instance) then we can't decide which one to execute. let me know your thoughts on this. Thanks
> 
> @SheikSena can we make it a string instead of list and let the user only select one of the options?

Yes, @cdwilhelm  we can do that but whatever action user selects that action will be executed automatically. for example, if user selects stop instances then all the instances will be stopped automatically. If this is ok with you? then I'll update the policy. Thanks

**Comment by cdwilhelm** [comment]
> Yes, @cdwilhelm we can do that but whatever action user selects that action will be executed automatically. for example, if user selects stop instances then all the instances will be stopped automatically. If this is ok with you? then I'll update the policy. Thanks

@SheikSena   yes, that is the intention of the automatic action param, to automatically stop or terminate the instances.  If they don't chose any of the options they can pick resource and actions to run.

File: operational/aws/long_running_instances/long_running_instances.pt (hunk 6 above)
Comment (SheikSena) [reply operational/aws/long_running_instances/long_running_instances.pt:256]: Done!

Comment (SheikSena) [reply operational/aws/long_running_instances/long_running_instances.pt:256]: Done!

File: operational/aws/long_running_instances/long_running_instances.pt (hunk 8 above)
Comment (SheikSena) [reply operational/aws/long_running_instances/long_running_instances.pt:51]: Done!

Comment (SheikSena) [reply operational/aws/long_running_instances/long_running_instances.pt:51]: Done!

File: operational/aws/long_running_instances/README.md (hunk 7 above)
Comment (SheikSena) [reply operational/aws/long_running_instances/README.md:20]: Done!

**Pull request merged**